### PR TITLE
[CARBONDATA-2776][CarbonStore] Support ingesting data from Kafka service

### PR DESCRIPTION
**Pull request by QiangCai**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

File: store/sql/pom.xml [hunk 1]
```diff
@@ -35,6 +36,33 @@
...
+        <exclusion>
+          <groupId>com.fasterxml.jackson.core</groupId>
+          <artifactId>*</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>com.amazonaws</groupId>
+      <artifactId>aws-java-sdk-s3</artifactId>
+      <version>	1.10.6</version>
```
Comment (xuchuanyin) [inline store/sql/pom.xml]: trim the useless spaces

File: pom.xml [hunk 2]
```diff
@@ -110,7 +110,7 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <snappy.version>1.1.2.6</snappy.version>
-    <hadoop.version>2.7.2</hadoop.version>
+    <hadoop.version>2.8.3</hadoop.version>
```
Comment (xuchuanyin) [inline pom.xml]: can we directly upgrade the hadoop version to 2.8.3 without any other changes?

File: store/sql/src/main/java/org/apache/carbondata/dis/DisProducer.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+        for (int j = 0; j < putRecordsResult.getRecords().size(); j++) {
+          PutRecordsResultEntry putRecordsRequestEntry1 = putRecordsResult.getRecords().get(j);
+          if (putRecordsRequestEntry1.getErrorCode() != null) {
+            LOGGER.error("[" + new String(
+                putRecordsRequestEntryList.get(j).getData().array(), Charset.defaultCharset())
+                + "] put failed, errorCode [" + putRecordsRequestEntry1.getErrorCode()
+                + "], errorMessage [" + putRecordsRequestEntry1.getErrorMessage() + "]");
+          } else {
+            LOGGER.info("[" + new String(
```
Comment (xuchuanyin) [inline store/sql/src/main/java/org/apache/carbondata/dis/DisProducer.java]: why not use stringbuilder or stringformat

**Comment by jackylk** [comment]
please rebase

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7539/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6293/

File: store/sql/src/main/java/org/apache/carbondata/dis/DisProducer.java (hunk 3 above)
Comment (QiangCai) [reply store/sql/src/main/java/org/apache/carbondata/dis/DisProducer.java]: ok

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7594/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7595/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6368/

**Comment by jackylk** [comment]
LGTM

**Comment by jackylk** [comment]
merging into carbonstore branch

**Pull request closed without merging**